local/setup.sh

```diff
@@ -11,113 +48,143 @@ RUN_CLIENT="${2:-1}"
-		--network ${NETWORK} \
-		${IMAGE_APISERVER}:${TAG}
+    echo 'Running cluster-registry api'
+    if container_exists "${CONTAINER_API}"; then
+        container_running "${CONTAINER_API}" && { docker stop "$CONTAINER_API" || die "Failed to stop cluster-registry api container $CONTAINER_API"; }
+        docker rm "${CONTAINER_API}" || die "Failed to remove cluster-registry api container $CONTAINER_API"
+    fi
+        docker run -d \
+            --name "${CONTAINER_API}" \
+            -p 8080:8080 \
+            -e AWS_REGION \
+            -e AWS_ACCESS_KEY_ID \
+            -e AWS_SECRET_ACCESS_KEY \
+            -e DB_AWS_REGION \
+            -e DB_ENDPOINT=http://"${CONTAINER_DB}":8000 \
+            -e DB_TABLE_NAME="${DB_TABLE_NAME}" \
+            -e DB_INDEX_NAME="${DB_INDEX_NAME}" \
+            -e OIDC_ISSUER_URL=http://"${CONTAINER_OIDC}" \
+            -e OIDC_CLIENT_ID \
+            -e SQS_AWS_REGION \
+            -e SQS_ENDPOINT=http://"${CONTAINER_SQS}":9324 \
+            -e SQS_QUEUE_NAME="${SQS_QUEUE_NAME}" \
+            -e API_RATE_LIMITER="${API_RATE_LIMITER}" \
+            -e LOG_LEVEL="${LOG_LEVEL}" \
+            -e API_HOST="${API_HOST}" \
+            -e K8S_RESOURCE_ID="${K8S_RESOURCE_ID}" \
+            -e API_TENANT_ID="${API_TENANT_ID}" \
+            -e API_CLIENT_ID="${API_CLIENT_ID}" \
+            -e API_CLIENT_SECRET="${API_CLIENT_SECRET}" \
+            -e API_AUTHORIZED_GROUP_ID="${API_AUTHORIZED_GROUP_ID}" \
+            --network "${NETWORK}" \
+            "${IMAGE_APISERVER}":"${TAG}" || die "Failed to create $CONTAINER_API container."
 fi
 
 if [[ "${RUN_CLIENT}" == 1 ]]; then
-	echo 'Running cluster-registry-client'
-	docker run -d \
-		--name ${CONTAINER_CLIENT} \
-		-v "${ROOT_DIR}/kubeconfig_client":/kubeconfig \
-		-e AWS_ACCESS_KEY_ID \
-		-e AWS_SECRET_ACCESS_KEY \
-		-e KUBECONFIG=/kubeconfig \
-		-e SQS_AWS_REGION \
-		-e SQS_ENDPOINT=http://${CONTAINER_SQS}:9324 \
-		-e SQS_QUEUE_NAME=${SQS_QUEUE_NAME} \
-		--network ${NETWORK} \
-		${IMAGE_CLIENT}:${TAG}
+    echo 'Running cluster-registry-client'
+    if container_exists "${CONTAINER_CLIENT}"; then
+        container_running "${CONTAINER_CLIENT}" && { docker stop "$CONTAINER_CLIENT" || die "Failed to stop cluster-registry-client container $CONTAINER_CLIENT"; }
+        docker rm "${CONTAINER_CLIENT}" || die "Failed to remove cluster-registry-client container $CONTAINER_CLIENT"
+    fi
+    docker run -d \
+        --name "${CONTAINER_CLIENT}" \
+        -v "${ROOT_DIR}/kubeconfig_client":/kubeconfig \
+        -e AWS_ACCESS_KEY_ID \
+        -e AWS_SECRET_ACCESS_KEY \
+        -e KUBECONFIG=/kubeconfig \
+        -e SQS_AWS_REGION \
+        -e SQS_ENDPOINT=http://"${CONTAINER_SQS}":9324 \
+        -e SQS_QUEUE_NAME="${SQS_QUEUE_NAME}" \
+        --network "${NETWORK}" \
+        "${IMAGE_CLIENT}":"${TAG}" || die "Failed to create $CONTAINER_CLIENT container."
 fi
 
 echo 'Local stack was set up successfully.'
+
+# vim: ts=4 sw=4 et
```

Comment:
what do you think of using https://editorconfig.org/ to store these options?

Reply:
:facepalm: I totally missed that! Let me move it there.